www/SQLitePlugin.js

```diff
@@ -221,33 +225,46 @@
               
             };
         };
-      })(this);
-      openerrorcb = (function(_this) {
-        return function() {
+      openerrorcb = function() {
+          failed = true;
           console.log('OPEN database: ' + _this.dbname + ' FAILED, aborting any pending transactions');
           if (!!error) {
             error(newSQLError('Could not open database'));
           }
           delete _this.openDBs[_this.dbname];
           _this.abortAllPendingTransactions();
         };
-      })(this);
       this.openDBs[this.dbname] = { 
         state: DB_STATE_INIT,
         connections: []
       };
 
-      var options = {};
-      for (var key in this.openargs) {
-        options[key] = this.openargs[key];
+      var copyOptions = function() {
+        var options = {};
+        var optionNames = Object.keys(_this.openargs);
+        for (var i = 0; i < optionNames.length; i++) {
```

Comment:
`for-in` shouldn't be used as-is because it can easily break when the object's prototype is extended. We can either do add a `hasOwnProperty` check or just iterate what `Object.keys` gives us.